v2/kafka-to-bigquery/src/main/java/com/google/cloud/teleport/v2/transforms/BigQueryDynamicDestination.java

```diff
@@ -58,8 +59,18 @@ public GenericRecord getDestination(ValueInSingleWindow<KV<GenericRecord, TableR
 
   @Override
   public TableDestination getTable(GenericRecord element) {
-    // tablename + record name (same across schemas) + schema id?
-    String bqQualifiedFullName = element.getSchema().getFullName().replace(".", "-");
+    Pattern sanitizePattern = Pattern.compile("[^\\w- ]", Pattern.UNICODE_CHARACTER_CLASS);
```

Comment:
`Pattern.compile` is a relatively expensive operation, it's usually done only once:

```
  private static final Pattern SANITIZE_PATTERN = Pattern.compile("[^\\w- ]", Pattern.UNICODE_CHARACTER_CLASS);

  @Override
  public TableDestination getTable(GenericRecord element) {
    String sanitizedNamespace =
        SANITIZE_PATTERN.matcher(element.getSchema().getNamespace()).replaceAll("-");
    ....
```

Reply:
done